BlueCapKit/Central/CentralManager.swift

```diff
@@ -310,11 +310,18 @@ public class CentralManager : NSObject, CBCentralManagerDelegate {
     }
     
     func didDiscoverPeripheral(_ peripheral: CBPeripheralInjectable, advertisementData: [String : Any], RSSI: NSNumber) {
-        let bcPeripheral: Peripheral
+        var bcPeripheral: Peripheral!
+
         if let discoveredPeripheral = _discoveredPeripherals[peripheral.identifier] {
-            bcPeripheral = discoveredPeripheral
-            bcPeripheral._RSSI = RSSI.intValue
-        } else {
+            do {
+                try discoveredPeripheral.updateDiscovery(peripheral, advertisements: advertisementData, RSSI: RSSI.intValue)
```

Comment:
I think the discovery future should be completed with an error if the CBPeripheral references do not match. We could also complete the future with the modified Peripheral.